### PR TITLE
fix: workspaceFolder variable replacement and docs

### LINKED ISSUE #21: using ${workspaceFolder}  in config throwing errors
Hi, 

I'm trying to use the extension with CLI usage with a cli and company central config that we provide to all our projects and repos via an npm package.   While trying to migrate from the deprecated alternative vscode plugin to this one since it has CLI support (thanks!) I could not get the documented behavior of using the workspaceFolder variable to run. 

My configuration in the workspace file is (not working)
```
...
"vale.core.useCLI": true,
"vale.valeCLI.path": "${workspaceFolder}/node_modules/.bin/commercetools-vale-bin",
 "vale.valeCLI.config": "${workspaceFolder}/node_modules/@commercetools-docs/writing-style/.vale.ini",
...
```
The developer console shows this error:
image

**When replacing `${workspaceFolder}` with the absolute path to my workspace the plugin works fine**.  So I think something' wrong with the way the binary is called.  It's failing already at startup when the config is read, so I can't tell whether further calls are OK or not. 

I saw that some of the code here is identical to the one in https://github.com/testthedocs/vscode-vale/blob/master/src/extension.ts (the deprecated alternative plugin), but this codebase is lacking the explicit handling of the variable like done in https://github.com/testthedocs/vscode-vale/blob/master/src/extension.ts#L213 

Looks like something was ported but not completely refactored maybe?

### PR DESCRIPTION
**Pull request by emmenko**
Fixes #21 

I tried to port the changes from the previous plugin.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,18 +80,30 @@ The extension offers a number of settings and configuration options (_Preference
 
 - `vale.server.lintContext` (default: `0`): Only lint the *active* portion of a document (as determined by the cursor position), allowing for efficient on-the-fly linting of large documents. There are three supported values: `-1` (applies to all files), `0` (disabled), `n` (applies to any file with `lines >= n`).
 
-- `vale.valeCLI.config` (default: `null`): Absolute path to a Vale configuration file. If not specified, the extension uses the default search process (relative to the current file).
+- `vale.valeCLI.config` (default: `null`): Absolute path to a Vale configuration file. Use the predefined [`${workspaceFolder}`](https://code.visualstudio.com/docs/editor/variables-reference#_predefined-variables) variable to reference configuration file from a custom location. (**NOTE**: On Windows you can use '/' and can omit `.cmd` in the path value.) If not specified, the extension uses the default search process (relative to the current file).
```
Comment (emmenko) [inline README.md:83]: We should allow to use the `workspaceFolder` here as well.

File: src/features/vsProvider.ts [hunk 2]
```diff
@@ -212,11 +221,14 @@ export default class ValeServerProvider implements vscode.CodeActionProvider {
...
-    this.readabilityStatus = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 100);
+    this.readabilityStatus =
+      vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 100);
     this.diagnosticCollection = vscode.languages.createDiagnosticCollection();
 
     this.useCLI = configuration.get('vale.core.useCLI', false);
-    this.stylesPath = await utils.getStylesPath(this.useCLI);
+    // this.stylesPath = await utils.getStylesPath({
+    //   usingCLI: this.useCLI,
+    // });
```
Comment (emmenko) [inline src/features/vsProvider.ts:231]: During activation, I don't see how we can get the information about the workspace folder, which is required to get the correct binary + config paths.

Is setting the `stylesPath` necessary at this point, given that it's also set on `runVale`?

**Comment by jdkato** [comment]
I appreciate the effort in this PR, but I think it contains too many unrelated style and refactoring changes.

Please see #23.

**Comment by emmenko** [comment]
Sure, whatever as long as we are able to fix it 🙂

PS: I would recommend to install prettier in this project, as otherwise contributing to it might result in unrelated formatting changes. For instance, my VSCode is configured to use prettier when saving/formatting.
Just a suggestion...

**Pull request closed without merging**